docs/en/observability/apm/apm-performance-diagnostic.asciidoc

```diff
@@ -0,0 +1,43 @@
+As a result APM agents connected to the affected APM Server will suffer from throttling and/or request timeout when shipping APM events. 
+
+To quickly identify possible issues try looking for similar error logs lines in APM Server logs:
+
+[source,json]
+----
+...
+{"log.level":"error","@timestamp":"2024-07-27T23:46:28.529Z","log.origin":{"function":"github.com/elastic/go-docappender/v2.(*Appender).flush","file.name":"v2@v2.2.0/appender.go","file.line":370},"message":"bulk indexing request failed","service.name":"apm-server","error":{"message":"flush failed (429): [429 Too Many Requests]"},"ecs.version":"1.6.0"}
+{"log.level":"error","@timestamp":"2024-07-27T23:55:38.612Z","log.origin":{"function":"github.com/elastic/go-docappender/v2.(*Appender).flush","file.name":"v2@v2.2.0/appender.go","file.line":370},"message":"bulk indexing request failed","service.name":"apm-server","error":{"message":"flush failed (503): [503 Service Unavailable]"},"ecs.version":"1.6.0"}
+...
+----
+
+To gain better insight into APM Server health and performance, consider enabling the monitoring feature by following the steps in <<apm-monitor-apm-self-install,Monitor a Fleet-managed APM Server>>.
+When enabled APM Server will additionally report a set of vital metrics to help you identify any performance degradation.
+
+Pay careful attention to the next metric fields:
+
+* `beats_stats.metrics.libbeat.output.events.active` that represents the number of buffered pending documents waiting for indexing;
+(_if this value is increasing rapidly it indicates {es} backpressure_)
+* `beats_stats.metrics.libbeat.output.events.acked` that represents the number of indexing operations that have completed successfully;
+* `beats_stats.metrics.libbeat.output.events.failed` that represents the number of indexing operations that failed, it includes all failures;
+(_if this value is increasing rapidly it indicates {es} backpressure_)
+* `beats_stats.metrics.libbeat.output.events.toomany` that represents the number of indexing operations that failed due to {es} responding with 429 Too many Requests;
+(_if this value is increasing rapidly it indicates {es} backpressure_)
+* `beats_stats.output.elasticsearch.bulk_requests.available` that represents the number of bulk indexers available for making bulk index requests;
+(_if this value is equal to 0 it indicates {es} backpressure_)
+* `beats_stats.output.elasticsearch.bulk_requests.completed` that represents the number of already completed bulk requests;
+* `beats_stats.metrics.output.elasticsearch.indexers.active` that represents the number of active bulk indexers that are concurrently processing batches;
+
+See https://www.elastic.co/guide/en/beats/metricbeat/current/exported-fields-beat.html[{metricbeat} documentation] for the full list of exported metric fields.
```

Comment:
@colleenmcginnis is there a way to link to the metricbeat docs on the same release version?

Reply:
Yep! 

```suggestion
See {metricbeat-ref}/exported-fields-beat.html[{metricbeat} documentation] for the full list of exported metric fields.
```

Reply:
updated